File: aula1/Hw1.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <string.h>
#include <stdbool.h>

/* run this program using the console pauser or add your own getch, system("pause") or input loop */


char * convertNumberIntoArray(unsigned int number) {
    unsigned int length = (int)(log10((float)number)) + 1;
    char * arr = (char *) malloc(length * sizeof(char)), * curr = arr;
    do {
        *curr++ = number % 10;
        number /= 10;
    } while (number != 0);
    return arr;
}

void quantoFind(int x, int y){
	char *a = convertNumberIntoArray(y);
	int i = 0;
	int j;
	for(j = 0; j < strlen(a); j++)
	if(x == a[j])
		i++;
	printf("\nQuantidade: %d", i);
}

void permutacao(int x1, int y1){
	char *x = convertNumberIntoArray(x1);
	char *y = convertNumberIntoArray(y1);
	int a = 0, b = 0, c = 0, d = 0;
	int i;
	for(i = 0; i < strlen(x); i++){
		a+=x[i];
		b+=y[i];
		c*=x[i];
		d*=y[i];
	}
	if((a == b) && (c == d))
		printf("\nÉ permutação");
	else
		printf("\nNão é permutação");
}

void exercicioC(int x1, int y1){
	char *x = convertNumberIntoArray(x1);
	char *y = convertNumberIntoArray(y1);
	int i, count, findIndex = 0;
	bool ok[strlen(y)];
	for (i = 0; i < strlen(x); i++){
		if(x[i] == y[0]){
			findIndex = i;
			ok[0] = true;
		}
	}
	if((findIndex + (strlen(y) - 1)) < strlen(x)){
		int j;
		for (j = findIndex; j < (findIndex + strlen(y) - 1); j++){
			if(x[j+1] == y[1+j-findIndex]){
				ok[1+j-findIndex] = true;
			}
		}
		int i;
		for (i = 0; i < strlen(y); i++){
			if(ok[i]){
				count++;
			}
		}
		if(count == strlen(y))
			printf("\nÉ parte");
		else
			printf("\nNão é parte");
    }
	else{
		printf("\nNão é parte");
	}
}

int main(int argc, char *argv[]) {
	int num, operacao, d;
	printf("Entre com o número\n");
	scanf("%i", &num);
	printf("\nEscolha a operação: \n");
	printf("\n1. Achar quantidade de dígitos");
	printf("\n2. Verificar se o número é permutação de outro número");
	printf("\n3. Procurar por ocorrencia\n");
	scanf("%i", &operacao);
	switch(operacao){
		case 1:
			printf("\nEntre com o elemento\n");
			scanf("%i", &d);
			char *a = convertNumberIntoArray(num);
			quantoFind(d, num);
		break;
		case 2:
			printf("\nEntre com o elemento\n");
			scanf("%i", &d);
			permutacao(num, d);
		break;
		case 3:
			printf("\nEntre com o elemento\n");
			scanf("%i", &d);
			exercicioC(num, d);
		break;
		default:
			printf("\nFim da execução\n");
	}
	return 0;
}

File: AED1/aula1/Hw1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package hw1;

import java.util.Scanner;

/**
 *
 * @author Patrick
 */
public class Hw1 {

    /**
     * @param args the command line arguments
     */
    
    public static int[] convertToArray(int x){
        String y = Integer.toString(x);
        int[] newX = new int[y.length()];
        for (int i = 0; i < y.length(); i++)
            newX[i] = y.charAt(i) - '0';
        return newX;
    }
    
    public static void quantoFind(int x, int y){
        int z[] = convertToArray(y);
        int i = 0;
        for(int j = 0; j < z.length; j++)
            if(x == z[j])
                i++;
        System.out.println("Quantidade: "+i);
    }
    
    public static void permutacao(int x1, int y1){
        int x[] = convertToArray(x1);
        int y[] = convertToArray(y1);
        int a = 0, b = 0, c = 0, d = 0;
        for(int i = 0; i < x.length; i++){
            a+=x[i];
            b+=y[i];
            c*=x[i];
            d*=y[i];
        }
        if((a == b) && (c == d))
            System.out.println("É permutação");
        else
            System.out.println("Não é permutação");
    }
    
    public static void exercicioC(int x1, int y1){
        int x[] = convertToArray(x1);
        int y[] = convertToArray(y1);
        int findIndex = 0;
        boolean ok[] = new boolean[y.length];
        int count = 0;
        for (int i = 0; i < x.length; i++){
            if(x[i] == y[0]){
                findIndex = i;
                ok[0] = true;
            }
        }
        if((findIndex + (y.length - 1)) < x.length){
            for (int i = findIndex; i < (findIndex + y.length - 1); i++){
                if(x[i+1] == y[1+i-findIndex]){
                    ok[1+i-findIndex] = true;
                }
            }
            for (int i = 0; i < ok.length; i++){
                if(ok[i]){
                    count++;
                }
            }
            if(count == y.length)
                System.out.println("É parte");
            else
                System.out.println("Não é parte");
        }
        else{
            System.out.println("Não é parte");
        }
    }
    
    public static void main(String[] args) {
        // TODO code application logic here
        System.out.println("Entre com o número");
        Scanner entrada = new Scanner(System.in);
        int num = entrada.nextInt();
        System.out.println("Escolha a operação: \n");
        System.out.println("1. Achar quantidade de dígitos");
        System.out.println("2. Verificar se o número é permutação de outro número");
        System.out.println("3. Procurar por ocorrencia");
        int operacao = entrada.nextInt();
        int d;
        switch(operacao){
            case 1:
                System.out.println("Entre com o elemento");
                d = entrada.nextInt();
                quantoFind(d, num);
                break;
            case 2:
                System.out.println("Entre com o elemento");
                d = entrada.nextInt();
                permutacao(num, d);
                break;
            case 3:
                System.out.println("Entre com o elemento");
                d = entrada.nextInt();
                exercicioC(num, d);
                break;
            default:
                System.out.println("Fim da execução");
        }
    }
    
}
